crud/select/plan.lua

```diff
@@ -48,6 +49,46 @@ local function get_index_for_condition(space_indexes, space_format, condition)
     end
 end
 
+local function extract_sharding_key_from_conditions(conditions, ddl_sharding_key, space_indexes, space_field_positions)
+    if ddl_sharding_key == nil then
+        return nil
+    end
+
+    -- If name is both valid index name and field name,
+    -- it is interpreted as index name.
+    local filled_fields = {}
+    for _, condition in ipairs(conditions) do
+        if condition.operator ~= compare_conditions.operators.EQ then
+            goto continue
+        end
+
+        local index = space_indexes[condition.operand]
+        if index ~= nil then
+            for i, part in ipairs(index.parts) do
+                filled_fields[part.fieldno] = condition.values[i]
+            end
```

Comment:
Please, remind me, whether we support partial keys here?

Reply:
As you know, I had some troubles with partial keys (https://github.com/tarantool/tarantool/issues/6655). I added a test case for partial index with last missing field.

Reply:
Let's consider the example:

tuple field names: `{'foo', 'bar', 'baz'}`
index_0: `{'foo', 'bar'}`
index_1: `{'baz', 'bar'}`

conditions: `{{'==', 'index_0', {1, 2}}, {'==', 'index_1', {3, nil}}}`

After processing of the first condition (EQ by index_0), `filled_fields` will be `{1, 2, nil}`. After processing of the second condition (partial EQ by index_1), it will be `{1, nil, 3}`. However it is know that the only tuple we can receive is `{1, 2, 3}`.

Correct?

Reply:
Thank you for notice. Fixed it, added several test cases.

Reply:
Well, the check against `nil` should work. Since it was not obvious neither for me, nor for you, it looks as the check that should have a comment in the code.

Reply:
Added

Reply:
Nice.